File: practice/gg1.py
# Palindrome Number Check
num = int(input("Enter a number: "))
rev = int(str(num)[::-1])

if num == rev:
    print(num, "is a palindrome")
else:
    print(num, "is not a palindrome")



# Fibonacci Series
n = int(input("Enter number of terms: "))

a, b = 0, 1
print("Fibonacci Series:")
for i in range(n):
    print(a, end=" ")
    a, b = b, a + b



# Armstrong Number Check
num = int(input("Enter a number: "))
digits = str(num)
power = len(digits)

total = sum(int(i)**power for i in digits)

if total == num:
    print(num, "is an Armstrong number")
else:
    print(num, "is not an Armstrong number")



# Palindrome String Check
text = input("Enter a string: ")
if text == text[::-1]:
    print("Palindrome String")
else:
    print("Not Palindrome")



# Vowel Count
text = input("Enter a string: ")
vowels = "aeiouAEIOU"
count = sum(1 for ch in text if ch in vowels)
print("Total vowels =", count)



# Multiplication Table
for i in range(1, 11):
    for j in range(1, 11):
        print(i*j, end="\t")
    print()



# Sum of Digits
num = input("Enter a number: ")
total = sum(int(d) for d in num)
print("Sum of digits =", total)




# Matrix Addition
import numpy as np

A = np.array([[1,2,3],
              [4,5,6],
              [7,8,9]])

B = np.array([[9,8,7],
              [6,5,4],
              [3,2,1]])

result = A + B
print((result))








A = [[1,2,3],
     [4,5,6],
     [7,8,9]]

B = [[9,8,7],
     [6,5,4],
     [3,2,1]]

result = [[0,0,0],
          [0,0,0],
          [0,0,0]]

for i in range(len(A)):
    for j in range(len(A[0])):
        result[i][j] = A[i][j] + B[i][j]

print("Result matrix:")
for row in result:
    print(row)

File: practice/Cloud/gg1.c
// Check if a number is palindrome or not
#include <stdio.h>
int main() {
    int n, temp, rev = 0, r;
    printf("Enter number: ");
    scanf("%d", &n);
    temp = n;
    while (n > 0) {
        r = n % 10;
        rev = rev * 10 + r;
        n = n / 10;
    }
    if (temp == rev) printf("%d is a Palindrome", temp);
    else printf("%d is a Not Palindrome", temp);
    return 0;
}


// Print Fibonacci series up to n terms
#include <stdio.h>
int main() {
    int n, a = 0, b = 1, c, i;
    printf("Enter n terms: ");
    scanf("%d", &n);
    for (i = 1; i <= n; i++) {
        printf("%d ", a);
        c = a + b;
        a = b;
        b = c;
    }
    return 0;
}


// Check if a number is Armstrong or not
#include <stdio.h>
#include <math.h>

int main() {
    int n, temp, r, sum = 0;
    printf("Enter number: ");
    scanf("%d", &n);

    temp = n;
    int digits = log10(n) + 1;  // simpler way to get digits

    while (temp > 0) {
        r = temp % 10;
        sum += pow(r, digits);
        temp /= 10;
    }

    (sum == n) ? printf("Armstrong") : printf("Not Armstrong");
    return 0;
}


// Check if a string is palindrome or not
#include <stdio.h>
#include <string.h>

int main() {
    char s[100];
    printf("Enter string: ");
    scanf("%s", s);

    int len = strlen(s);
    int i;
    int isPalindrome = 1;

    for (i = 0; i < len/2; i++) {
        if (s[i] != s[len - 1 - i]) {
            isPalindrome = 0;
            break;
        }
    }

    if (isPalindrome)
        printf("Palindrome");
    else
        printf("Not Palindrome");

    return 0;
}


// Calculate sum of digits of a number
#include <stdio.h>
int main() {
    int n, sum = 0, r;
    printf("Enter number: ");
    scanf("%d", &n);
    while (n > 0) {
        r = n % 10;
        sum += r;
        n /= 10;
    }
    printf("Sum = %d", sum);
    return 0;
}


// Add two matrices
#include <stdio.h>
int main() {
    int a[10][10], b[10][10], sum[10][10];
    int r, c, i, j;

    printf("Enter rows and cols: ");
    scanf("%d %d", &r, &c);

    printf("Enter matrix A:\n");
    for(i=0;i<r;i++)
        for(j=0;j<c;j++)
            scanf("%d",&a[i][j]);

    printf("Enter matrix B:\n");
    for(i=0;i<r;i++)
        for(j=0;j<c;j++)
            scanf("%d",&b[i][j]);

    for(i=0;i<r;i++)
        for(j=0;j<c;j++)
            sum[i][j] = a[i][j] + b[i][j];

    printf("Sum matrix:\n");
    for(i=0;i<r;i++){
        for(j=0;j<c;j++)
            printf("%d ", sum[i][j]);
        printf("\n");
    }
    return 0;
}


// Read and print a paragraph
#include <stdio.h>

int main() {
    char paragraph[500]; // Buffer to store the input

    printf("Enter a paragraph:\n");
    
    // Read a line of text into the 'paragraph' buffer
    fgets(paragraph, sizeof(paragraph), stdin);
    
    printf("\nInput received: %s\n", paragraph);

    return 0;
}


// Print a right-angled triangle pattern of stars
#include <stdio.h>
int main() {
    int n,i,j;
    scanf("%d",&n);

    for(i=1;i<=n;i++){
        for(j=1;j<=i;j++){
            printf("*");
        }
        printf("\n");
    }

    return 0;
}
// *
// **
// ***
// ****
// *****



// Print an inverted right-angled triangle pattern of stars
#include <stdio.h>
int main() {
    int n,i,j;
    scanf("%d",&n);

    for(i=n;i>=1;i--){
        for(j=1;j<=i;j++){
            printf("*");
        }
        printf("\n");
    }

    return 0;
}
// *****
// ****
// ***
// **
// *



// Print a pyramid pattern of stars
#include <stdio.h>
int main() {
    int n,i,j;
    scanf("%d",&n);

    for(i=1;i<=n;i++){
        // print spaces
        for(j=1;j<=n-i;j++){
            printf(" ");
        }
        // print stars
        for(j=1;j<=2*i-1;j++){
            printf("*");
        }
        printf("\n");
    }

    return 0;
}
//     *
//    ***
//   *****
//  *******
// *********



// Print a right-aligned triangle pattern of stars
#include <stdio.h>
int main() {
    int n,i,j;
    scanf("%d",&n);

    for(i=1;i<=n;i++){
        for(j=1;j<=n-i;j++){
            printf(" ");   // spaces first
        }
        for(j=1;j<=i;j++){
            printf("*");   // then stars
        }
        printf("\n");
    }
    return 0;
}
//     *
//    **
//   ***
//  ****
// *****
